### DISCUSSION TITLE
Whether the pinia instance must be passed in to use onserverprefetch under ssr，and whether the failure to pass will cause status contamination

### DISCUSSION THREAD

**Discussion by guaqiu5** [vuejs/pinia#1991] [Help and Questions] [answered]
I see in doc:
<img width="871" alt="image" src="https://user-images.githubusercontent.com/57245152/218120019-903f3727-d48c-4c34-b637-0510e5c28ca3.png">

That's what I'm using at the moment:
```
<script setup lang="ts">
 import { useUserStore } from '~/store/user'
 const userStore = useUserStore()
onServerPrefetch(async () => {
  // may change some state in store
    await userStore.getUserInfo()
  })
</scirpt>
```

Will not passing down pinia instances like " useUserStore(pinia)" from ssr cause state contamination？

**Comment by posva** [comment, marked as answer]
Not in this case because you are within `setup()`. Passing the pinia on SSR is only necessary outside of setup to ensure the correct pinia context is used
